lib/ansible/plugins/action/copy.py

```diff
@@ -186,6 +186,12 @@ def _recurse(topdir, rel_offset, parent_dirs, rel_base=u''):
 
 class ActionModule(ActionBase):
 
+    def _remove_action_keys(self, module_args):
```

Comment:
Now that I can see this from a higher level, I think it makes sense to combine this filtering with creation of the args.

So this function becomes:
``` python
def _create_remote_copy_args(self, module_args):
    return dict((k, v) for k, v in module_args if k not in ('content', 'decrypt'))
```

And the calls to this get combined with the lines like:
``` python
- new_module_args = self._task.args.copy()
+ new_module_args = self._create_remote_copy_args(self._task.args)
```

Reply:
i was going even more generic, as this issue arises from 'remote file' args not matching